numpy/lib/function_base.py

```diff
@@ -307,10 +307,12 @@ def average(a, axis=None, weights=None, returned=False):
     average, [sum_of_weights] : array_type or double
         Return the average along the specified axis. When returned is `True`,
         return a tuple with the average as the first element and the sum
-        of the weights as the second element. The return type is `Float`
-        if `a` is of integer type, otherwise it is of the same type as `a`.
-        `sum_of_weights` is of the same type as `average`.
-
+        of the weights as the second element.`sum_of_weights` is of the
+        same type as `average`. The result type is the type of lowest precision
+        capable of representing values of both `a` and `weights` or 'float64'
+        if that type would be integral. Otherwise, if `a` is non integral,
```

Comment:
This is taken from [my comment here](https://github.com/numpy/numpy/pull/11355#issuecomment-398299375). Unfortunately, that's not how it works - `a = int8, weight = float32`, which gives a result of `float64`

Reply:
@eric-wieser You're right. I'll give it a try:

- If `weights` is `None`, the result type will be that of `a`, or `float64` if `a` is integral.
- If `weights != None` and `a` is *not* integral, the result type will be the type of lowest precision capable of representing values of both `a` and `weights`.
- If `weights != None` and `a` *is* integral, the previous rule applies, but the result type will be at least `float64`.

Hope I got it right. If anybody comes up with a more concise description, that would be great.